### PR TITLE
Add el9 selinux rpm

### PR DESCRIPTION
**Pull request by galal-hussein**
#### Proposed Changes ####

- Since we added el9 to k3s-selinux we should distinctly specify it in the install script.
- Added also rocky-9 install test to the unit tests using "testing channel" until k3s selinux is released as stable or rpm with the new changes and distros

#### Types of Changes ####

Install script

#### Linked Issues ####

- https://github.com/k3s-io/k3s/issues/7307

### CODE REVIEW

**Review by brandond — APPROVED** [review]

**Review by briandowns — APPROVED** [review]

File: install.sh [hunk 1]
```diff
@@ -518,7 +518,7 @@ setup_selinux() {
         rpm_target=el7
         rpm_site_infix=centos/7
         package_installer=yum
-    elif [ "${VERSION_ID%%.*}" = "8" ]; then
+    elif [ "${VERSION_ID%%.*}" = "8" ] || [ "${VERSION_ID%%.*}" = "37" ]; then
```
Comment (brandond) [inline install.sh:521]: is Fedora 37 the only one that needs the EL8 packages, instead of EL7 or EL9?

Comment (galal-hussein) [reply install.sh:521]: as far as I can see yes, it needs el8

Comment (itamarjp) [reply install.sh:521]: I recommend >= "37" so it can  work for future releases of fedora.

Comment (brandond) [reply install.sh:521]: Well I think 38 and newer need EL9 packages, and 36 needs EL7? I'm not sure, but some clarification would be good.

**Review by brandond — APPROVED** [review]

**Comment by galal-hussein** [comment]
I added a new modification to the PR to fix a problem with the upgrade:

The problem:

- el9 that has old container-selinux (2.189) and k3s-selinux policy
- upon upgrade the container-selinux to (2.205) it fails to upgrade because of existing k3s-selinux policy that exist in place.

To fix that we should either disable or remove k3s-selinux policy before upgrading container-selinux.

File: install.sh [hunk 2]
```diff
@@ -596,9 +596,12 @@ EOF
         sle)
             rpm_installer="zypper --gpg-auto-import-keys"
             if [ "${TRANSACTIONAL_UPDATE=false}" != "true" ] && [ -x /usr/sbin/transactional-update ]; then
+                transactional_update_run="transactional-update --no-selfupdate -d run"
                 rpm_installer="transactional-update --no-selfupdate -d run ${rpm_installer}"
                 : "${INSTALL_K3S_SKIP_START:=true}"
             fi
+            # create the /var/lib/rpm-state in SLE systems to fix the prein selinux macro
+                ${transactional_update_run} mkdir -p /var/lib/rpm-state
```
Comment (brandond) [inline install.sh:604]: nit: consistent indentation
```suggestion
            ${transactional_update_run} mkdir -p /var/lib/rpm-state
```

**Review by brandond — APPROVED** [review]

**Pull request merged**